atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -286,29 +293,59 @@ public synchronized void shutdown() {
 
     public enum SweepOutcome {
         SUCCESS, NOTHING_TO_SWEEP, DISABLED, UNABLE_TO_ACQUIRE_LOCKS,
-        NOT_ENOUGH_DB_NODES_ONLINE, TABLE_DROPPED_WHILE_SWEEPING, ERROR
+        NOT_ENOUGH_DB_NODES_ONLINE, TABLE_DROPPED_WHILE_SWEEPING, ERROR,
+        SHUTDOWN, FATAL
     }
 
     private class SweepOutcomeMetrics {
         private final MetricsManager metricsManager = new MetricsManager();
         private final SlidingTimeWindowReservoir reservoir;
 
+        private boolean shutdown;
+        private boolean fatal;
+
         SweepOutcomeMetrics() {
             Arrays.stream(SweepOutcome.values()).forEach(outcome ->
                     metricsManager.registerMetric(BackgroundSweeperImpl.class, outcome.name(),
                             () -> getOutcomeCount(outcome))
             );
             reservoir = new SlidingTimeWindowReservoir(60L, TimeUnit.SECONDS);
+            shutdown = false;
+            fatal = false;
         }
 
         private Long getOutcomeCount(SweepOutcome outcome) {
+            if (outcome == SweepOutcome.SHUTDOWN) {
+                return shutdown ? 1L : 0L;
+            }
+            if (outcome == SweepOutcome.FATAL) {
+                return fatal ? 1L : 0L;
+            }
+
             return Arrays.stream(reservoir.getSnapshot().getValues())
                     .filter(l -> l == outcome.ordinal())
                     .count();
         }
 
         void registerOccurrenceOf(SweepOutcome outcome) {
+            if (outcome == SweepOutcome.SHUTDOWN) {
+                shutdown();
+                return;
+            }
+            if (outcome == SweepOutcome.FATAL) {
+                fatal();
+                return;
+            }
+
             reservoir.update(outcome.ordinal());
         }
+
+        public void shutdown() {
```

Comment:
We should inline `shutdown()` and `fatal()` as these new APIs to SweepOutcomeMetrics could be confusing. I see this as something I register my outcome with and then it reports the last outcome when I call `getOutcome`.

Reply:
Yeah, I wrote this the other way round - `shutdown()` and `fatal()` were separate things so it made sense, but then I reused the SweepOutcome to make the metric creation easy and so had to guard against someone passing them to `registerOccurenceOf()`.

I'll inline everything as it should reduce API size.